rustls-mbedcrypto-provider/src/kx.rs

```diff
@@ -28,34 +30,46 @@ use rustls::ffdhe_groups;
 use rustls::ffdhe_groups::FfdheGroup;
 use rustls::Error;
 use rustls::NamedGroup;
-/// A key-exchange group supported by *mbedtls*.
+/// An EC key-exchange group supported by *mbedtls*.
 ///
 /// All possible instances of this type are provided by the library in
 /// the `ALL_KX_GROUPS` array.
-struct KxGroup {
+pub struct KxGroup<T: RngCallback> {
     /// The IANA "TLS Supported Groups" name of the group
     name: NamedGroup,
 
-    /// The corresponding [`agreement::Algorithm`]
+    /// The corresponding agreement algorithm
     agreement_algorithm: &'static agreement::Algorithm,
+
+    /// Callback to produce RNGs when needed
+    rng_provider: fn() -> Option<T>,
+}
+
+impl<T: RngCallback> KxGroup<T> {
+    /// Create a new [`KxGroup`] with given RNG provider callback.
+    pub const fn with_rng_provider<F: RngCallback>(&self, rng_provider: fn() -> Option<F>) -> KxGroup<F> {
+        KxGroup { rng_provider, name: self.name, agreement_algorithm: self.agreement_algorithm }
+    }
 }
 
-impl fmt::Debug for KxGroup {
+impl<T: RngCallback> fmt::Debug for KxGroup<T> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         write!(f, "{:?}", self.name)
     }
 }
 
-impl SupportedKxGroup for KxGroup {
+impl<T: RngCallback + 'static> SupportedKxGroup for KxGroup<T> {
```

Comment:
I was adding this for making compiler happy. Let me double check it now again

Reply:
Without this, compiler will throw error:
```
error[E0310]: the parameter type `T` may not live long enough
  --> rustls-mbedcrypto-provider/src/kx.rs:77:12
   |
77 |           Ok(Box::new(KeyExchange {
   |  ____________^
78 | |             name: self.name,
79 | |             agreement_algorithm: self.agreement_algorithm,
80 | |             priv_key,
81 | |             pub_key: OnceLock::new(),
82 | |             rng_provider: self.rng_provider,
83 | |         }))
   | |          ^
   | |          |
   | |__________the parameter type `T` must be valid for the static lifetime...
   |            ...so that the type `T` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound
   |
61 | impl<T: RngCallback + 'static> SupportedKxGroup for KxGroup<T> {
```

Reply:
Actually this is strange, since `rng_provider` is function pointer.
Do you idea idea about how to fix it?

Reply:
I think this is fine. ` + 'static` just ensures that T does not contains reference.

Reply:
Oh I see. It's because the `rng_provider` is also used in the active `KeyExchange`. Yeah this is fine.